File: api/googleTranslate.js
const { Router } = require("express");
const { TranslationServiceClient } = require("@google-cloud/translate").v3beta1;
const { FetchError } = require("node-fetch");
const { withAsyncErrorHandling, ApiError } = require("./errorHandling");

const router = Router();

router.get(
    "/googleTranslate/:word",
    withAsyncErrorHandling(async (req, res) => {
        const translationClient = new TranslationServiceClient({
            credentials: {
                client_email: process.env.GOOGLE_AUTH_CLIENT_EMAIL,
                private_key: process.env.GOOGLE_AUTH_PRIVATE_KEY
            }
        });

        try {
            const [response] = await translationClient.translateText({
                parent: translationClient.locationPath(
                    "norsk-dictionary-1563830111515",
                    "global"
                ),
                contents: [req.params.word],
                mimeType: "text/plain",
                sourceLanguageCode: "nb-NO",
                targetLanguageCode: "en-US"
            });

            res.json(response.translations[0].translatedText);
        } catch (err) {
            if (err instanceof FetchError) {
                throw new ApiError(503);
            } else {
                throw err;
            }
        }
    })
);

module.exports = router;

File: server/server/googletranslate.py
import google
from google.oauth2 import service_account
from google.cloud import translate_v2 as translate
import os

from server import app, ApiError


@app.route("/api/googleTranslate/<word>")
def google_translate(word):
    try:
        credentials = service_account.Credentials.from_service_account_info(
            {
                "token_uri": "https://oauth2.googleapis.com/token",
                "client_email": os.getenv("GOOGLE_AUTH_CLIENT_EMAIL"),
                "private_key": os.getenv("GOOGLE_AUTH_PRIVATE_KEY"),
            }
        )

        client = translate.Client(credentials=credentials)
        response = client.translate(word, source_language="no")
    except (ValueError, google.auth.exceptions.RefreshError) as e:
        raise ApiError(500) from e
    except google.auth.exceptions.TransportError as e:
        raise ApiError(503) from e

    return response["translatedText"]
